public/components/overview/mitre/components/techniques/techniques.tsx

```diff
@@ -177,29 +189,73 @@ export const Techniques = withWindowSize(class Techniques extends Component {
     }
   }
 
+  async getMitreTechniques (params) {
+    this.setState({ isSearching: true })
+    return await WzRequest.apiReq("GET", "/mitre/techniques", { params: params })
+    .then((res) => res)
+    .catch((err) => {
+       this.showToast(
+         'danger',
+         'Error',
+         `Mitre techniques could not be fetched: ${err}`,
+         3000
+       );
+      []
+    });
+  }
+
+  async buildMitreTechniquesFromApi () {
+    const params = { limit: 500 };
+    const output = await this.getMitreTechniques(params)
+    const totalItems = (((output || {}).data || {}).data || {}).total_affected_items;
+    let mitreTechniques = [];
+    if (totalItems && output.data && output.data.data && totalItems > 500) {
+      params.offset = 0;
+      mitreTechniques.push(...output.data.data.affected_items);
+      while (mitreTechniques.length < totalItems && params.offset < totalItems) {
+        params.offset += params.limit;
+        const tmpData = await this.getMitreTechniques(params)
+        mitreTechniques.push(...tmpData.data.data.affected_items);
+      }
+    }
+    this.setState({ mitreTechniques: mitreTechniques, isSearching: false })
```

Comment:
When the `totalItems` is lower than 500, the `mitreTechniques` variable seems to be an empty array. Is this a bug?

Reply:
Yes, is a bug. Before validation I have to save `output.data.data.affected_items` first.